src/sentry/models/organizationmembermapping.py

```diff
@@ -54,4 +55,11 @@ class Meta:
             ("organization_id", "organizationmember_id"),
         )
 
+    def save(self, *args, **kwds):
+        with transaction.atomic(), in_test_psql_role_override("postgres"):
+            if self.user and self.id is None:
```

Comment:
This will do an extra query most of the time. Could we check for user_id instead?